### PR TITLE
feat: add okx bitcoin connector

### PR DESCRIPTION
**Pull request by zoruka**
# Description

Adds Bitcoin OKX Wallet connector.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-1687

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: apps/laboratory/src/utils/BitcoinUtil.ts [hunk 1]
```diff
@@ -113,6 +116,31 @@ export const BitcoinUtil = {
...
+    } else if (bitcoinPSBTUtils.isP2PKH(output)) {
+      return bitcoin.payments.p2pkh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2WPKH(output)) {
+      return bitcoin.payments.p2wpkh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2WSHScript(output)) {
+      return bitcoin.payments.p2wsh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2SHScript(output)) {
+      return bitcoin.payments.p2sh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2TR(output)) {
+      return bitcoin.payments.p2tr({ output, network })
```
Comment (tomiir) [inline apps/laboratory/src/utils/BitcoinUtil.ts:140]: maybe nit: can we do a switch or even an object mapping for this? 
eg
```
switch {
   case(bitcoinPSBTUtils.isP2MS(output)):
      ...
}
```

Not sure this works in js but wall of if looks ugly

File: packages/adapters/bitcoin/package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "@wallet-standard/app": "1.0.1",
     "@wallet-standard/base": "1.0.1",
     "@walletconnect/universal-provider": "2.17.0",
+    "bitcoinjs-lib": "7.0.0-rc.0",
```
Comment (tomiir) [inline packages/adapters/bitcoin/package.json:35]: do we need the rc version for this?

File: packages/adapters/bitcoin/src/connectors/OKXConnector.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  public readonly id = 'OKX'
+  public readonly name = 'OKX Wallet'
+  public readonly chain = 'bip122'
+  public readonly type = 'ANNOUNCED'
+  public readonly imageUrl =
+    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAJDSURBVHgB7Zq9jtpAEMfHlhEgQLiioXEkoAGECwoKxMcTRHmC5E3IoyRPkPAEkI7unJYmTgEFTYwA8a3NTKScLnCHN6c9r1e3P2llWQy7M/s1Gv1twCP0ej37dDq9x+Zut1t3t9vZjDEHIiSRSPg4ZpDL5fxkMvn1cDh8m0wmfugfO53OoFQq/crn8wxfY9EymQyrVCqMfHvScZx1p9ls3pFxXBy/bKlUipGPrVbLuQqAfsCliq3zl0H84zwtjQrOw4Mt1W63P5LvBm2d+Xz+YzqdgkqUy+WgWCy+Mc/nc282m4FqLBYL+3g8fjDxenq72WxANZbLJeA13zDX67UDioL5ybXwafMYu64Ltn3bdDweQ5R97fd7GyhBQMipx4POeEDHIu2LfDdBIGGz+hJ9CQ1ABjoA2egAZPM6AgiCAEQhsi/C4jHyPA/6/f5NG3Ks2+3CYDC4aTccDrn6ojG54MnEvG00GoVmWLIRNZ7wTCwDHYBsdACy0QHIhiuRETxlICWpMMhGZHmqS8qH6JLyGegAZKMDkI0uKf8X4SWlaZo+Pp1bRrwlJU8ZKLIvUjKh0WiQ3sRUbNVq9c5Ebew7KEo2m/1p4jJ4qAmDaqDQBzj5XyiAT4VCQezJigAU+IDU+z8vJFnGWeC+bKQV/5VZ71FV6L7PA3gg3tXrdQ+DgLhC+75Wq3no69P3MC0NFQpx2lL04Ql9gHK1bRDjsSBIvScBnDTk1WrlGIZBorIDEYJj+rhdgnQ67VmWRe0zlplXl81vcyEt0rSoYDUAAAAASUVORK5CYII='
```
Comment (tomiir) [inline packages/adapters/bitcoin/src/connectors/OKXConnector.ts:14]: can we get a proper image from design?

File: packages/adapters/bitcoin/src/connectors/OKXConnector.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    const from = (await this.wallet.getAccounts())[0]
+
+    if (!from) {
+      throw new Error('No account available')
+    }
+
+    const value = (Number(params.amount) / 10 ** network.nativeCurrency.decimals).toString()
+
```
Comment (tomiir) [inline packages/adapters/bitcoin/src/connectors/OKXConnector.ts:73]: I think this is a bit succeptible to breaking. can we do a utils to parse the value? 
something like ethers does with `parseUnit`

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 1 above)
Comment (zoruka) [reply apps/laboratory/src/utils/BitcoinUtil.ts:140]: As far as I know this is not possible syntax wise, `switch` expects a value to match the value in each `case`. As we need to check through functions I don't think it is possible to use `switch` in this case.

**Review by tomiir — APPROVED** [review]

File: packages/adapters/bitcoin/src/connectors/OKXConnector.ts (hunk 3 above)
Comment (zoruka) [reply packages/adapters/bitcoin/src/connectors/OKXConnector.ts]: Pushed a change to get the icon from the extension

**Pull request merged**